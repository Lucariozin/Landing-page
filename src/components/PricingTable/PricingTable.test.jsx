import { PricingTable } from '.';
import { render, screen } from '@testing-library/react';

describe('<PricingTable />', () => {
  it('should render PricingTable correctly', () => {
    const { container } = render(<PricingTable />);

    const table = container.querySelector('table');

    expect(screen.getByText(/Pricing table/i));
    expect(table).toBeInTheDocument();
    expect(screen.getAllByText(/Title/i)).toHaveLength(5);
    expect(screen.getAllByText(/Content/i)).toHaveLength(15);
  });

  it('should to match snapshot', () => {
    const { container } = render(<PricingTable />);
    expect(container).toMatchSnapshot();
  });
});
