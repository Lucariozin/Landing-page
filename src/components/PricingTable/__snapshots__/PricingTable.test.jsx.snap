// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PricingTable /> should to match snapshot 1`] = `
.c0 {
  margin-top: 3rem;
  margin-bottom: 5rem;
  overflow: hidden;
  overflow-x: auto;
}

.c2 {
  text-align: left;
  font-style: italic;
  font-size: 1.6rem;
  margin-bottom: 1rem;
}

.c1 {
  min-width: 65rem;
  width: 100%;
  border-collapse: collapse;
  font-size: 2rem;
}

.c3 {
  background-color: var(--light-gray-color);
}

.c6 {
  background-color: var(--light-gray-color);
}

.c4 {
  text-align: left;
  padding: 1rem;
  border: 0.1rem solid var(--light-gray-color);
}

.c5 {
  padding: 1rem;
  border: 0.1rem solid var(--light-gray-color);
}

<div>
  <div
    class="c0"
  >
    <table
      class="c1"
    >
      <caption
        class="c2"
      >
        Pricing table
      </caption>
      <thead
        class="c3"
      >
        <tr
          class=""
        >
          <th
            class="c4"
          >
            Title 1
          </th>
          <th
            class="c4"
          >
            Title 2
          </th>
          <th
            class="c4"
          >
            Title 3
          </th>
          <th
            class="c4"
          >
            Title 4
          </th>
          <th
            class="c4"
          >
            Title 5
          </th>
        </tr>
      </thead>
      <tbody
        class=""
      >
        <tr
          class=""
        >
          <td
            class="c5"
          >
            Content 1
          </td>
          <td
            class="c5"
          >
            Content 2
          </td>
          <td
            class="c5"
          >
            Content 3
          </td>
          <td
            class="c5"
          >
            Content 4
          </td>
          <td
            class="c5"
          >
            Content 5
          </td>
        </tr>
        <tr
          class=""
        >
          <td
            class="c5"
          >
            Content 1
          </td>
          <td
            class="c5"
          >
            Content 2
          </td>
          <td
            class="c5"
          >
            Content 3
          </td>
          <td
            class="c5"
          >
            Content 4
          </td>
          <td
            class="c5"
          >
            Content 5
          </td>
        </tr>
        <tr
          class=""
        >
          <td
            class="c5"
          >
            Content 1
          </td>
          <td
            class="c5"
          >
            Content 2
          </td>
          <td
            class="c5"
          >
            Content 3
          </td>
          <td
            class="c5"
          >
            Content 4
          </td>
          <td
            class="c5"
          >
            Content 5
          </td>
        </tr>
      </tbody>
      <tfoot
        class="c6"
      >
        <tr
          class=""
        >
          <td
            class="c5"
          />
          <td
            class="c5"
          />
          <td
            class="c5"
          />
          <td
            class="c5"
          />
          <td
            class="c5"
          >
            Footer
          </td>
        </tr>
      </tfoot>
    </table>
  </div>
</div>
`;
